lib/wda/webdriveragent.js

```diff
@@ -76,10 +78,10 @@ class WebDriverAgent {
 
   async launch (sessionId) {
     if (this.webDriverAgentUrl) {
-      log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
+      log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl.href}'`);
       this.url = this.webDriverAgentUrl;
       this.setupProxies(sessionId);
-      return this.webDriverAgentUrl;
```

Comment:
it's better to keep it since the returned result is used to cache wda status in the driver:

```
     let wdaStatus = null;
      try {
        wdaStatus = await this.wda.launch(sessionId, realDevice);
      } catch (err) {
        this.logEvent('wdaStartFailed');
        let errorMsg = `Unable to launch WebDriverAgent because of xcodebuild failure: "${err.message}".`;
        if (this.isRealDevice()) {
          errorMsg += ` Make sure you follow the tutorial at ${WDA_REAL_DEV_TUTORIAL_URL}. ` +
                      `Try to remove the WebDriverAgentRunner application from the device if it is installed ` +
                      `and reboot the device.`;
        }
        await quitAndUninstall(errorMsg);
      }
....
```